src/pause_fade.c

```diff
@@ -0,0 +1,103 @@
+#include "task.h"
+#include "main.h"
+#include "pause_fade.h"
+
+void CreatePauseFade(s8 arg0, u32 arg1) {
+    struct Task* state = TaskCreate(PauseFadeMain, 4, 0x100, 4, NULL);
+    struct PauseCommon *fade, *fade2;
+
+    if (state->unk12 & 0x10) {
+        fade = (struct PauseCommon*)(EWRAM_START + (state->unk6 << 2));
+    }
+    else {
+        fade = (struct PauseCommon*)(IWRAM_START + state->unk6);
+    }
+    fade2 = fade;
+
+    if ((arg0 << 0x18) > 0) {
```

Comment:
the shift is only for sign extension, so I guess you don't need it here for matching

same comment on other instances below